src/sensor_motion.py

```diff
@@ -0,0 +1,23 @@
+import RPi.GPIO as GPIO
+from time import sleep
+
+GPIO.setmode(GPIO.BCM)
+GPIO.setwarnings(False)
+pir = 16
+GPIO.setup(pir, GPIO.IN)
+
+print('Initializing Sensor...')
+sleep(2)
+
+def getMotion():
+    if GPIO.input(pir):
+        # motion detected
+        return True
+    
+    elif not GPIO.input(pir):
+        # no motion detected
+        return False
+
+    elif KeyboardInterrupt: 
+        GPIO.cleanup() # reset GPIO
+        print("Program ended")
```

Comment:
Wrap with the `try-catch`.
Store the `GPIO.input(dir)` to the variable.

Change `elif KeyboardInterrupt` block to `except KeyboardInterrupt`.
```python
elif KeyboardInterrupt: 
    GPIO.cleanup() # reset GPIO
    print("Program ended")
```

Also, Is there any chance that `GPIO.input` will throw an error?

Reply:
I don't know if GPIO.input could throw an error, but I assume its possible